var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -315,60 +315,61 @@ def configure_args(self):
 
     # run configure/make/make(install) for the nvptx-none target
     # before running the host compiler phases
-    @when('+nvptx')
     @run_before('configure')
     def nvptx_install(self):
         spec = self.spec
```

Comment:
If the `@when` clause isn't working, could you return here if `not spec.satisfies('+nvptx')` (vs. indenting everything below inside of an `if`)?

Reply:
Not without defining "spec" in some global context, which I was afraid to do.  It was simplest to avoid all of the logic in nvptx_install using the local definition present in the subroutine.  Feel free to improve on my solution and I'll just close this PR ...

Reply:
I'm saying if you inserted (at precisely this line), the following

```
if not spec.satisfies('+nvptx'):
    return
```

then you wouldn't have to indent everything below (i.e. I think the scope at which you are resolving it is good, just requesting an inverted if statement).

Reply:
Ok, I'll update the PR ...